src/AWS/CRT/Options.php

```diff
@@ -5,15 +5,46 @@
  */
 namespace AWS\CRT;
 
+final class OptionValue {
+    private $value;
+    function __construct($value) {
+        $this->value = $value;
+    }
+
+    public function asObject() {
+        return $this->value;
+    }
+
+    public function asMixed() {
+        return $this->value;
+    }
+
+    public function asInt() {
+        return is_null($this->value) ? 0 : (int)$this->value;
+    }
+
+    public function asBool() {
+        return boolval($this->value);
+    }
+
+    public function asString() {
+        return !empty($this->value) ? strval($this->value) : "";
+    }
+
+    public function asArray() {
+        return is_array($this->value) ? $this->value : [];
```

Comment:
shouldn't an as array function at least add the value to the array instead of returning an empty one?

Reply:
I guess if it's non-null, sure!